a11y-input/test/a11y-input.test.js

```diff
@@ -0,0 +1,127 @@
+  });
+});
+
+/**
+ * Testing Shadow DOM
+ */
+describe('a11y input', () => {
+  it(`has a static Shadow DOM`, async () => {
+    const el = /** @type {A11yInput} */ (await fixture(html`
+      <a11y-input></a11y-input>
+    `));
+
+    // ref: @open-wc chai plugin https://open-wc.org/testing/semantic-dom-diff.html
+    /**
+     * manual snapshot testing...
+     *
+     */
+    // expect(el).shadowDom.to.equal(`
+    //   <slot name="label"></slot>
+    //   <slot name="input"></slot>
+    // `);
+
+    /**
+     * semi/automatic snapshots using the `Karma` plugin
+     */
+    expect(el).shadowDom.to.equalSnapshot();
+  });
+});
+
+/**
+ * Testing Light DOM
+ */
+describe('a11y input', () => {
+  it('has 1 input and 1 label in the Light DOM', async () => {
+    const el = /** @type {A11yInput} */ (await fixture(html`
+      <a11y-input .label=${'foo'}></a11y-input>
+    `));
+
+    // debugger;
+    expect(el).lightDom.to.equal(`
+      <label slot="label">foo</label>
+      <input slot="input">
+    `);
+  });
+});
+
+/**
+ * Testing getter/setter
+ */
+describe('ally input', () => {
+  it('can set/get the input value directly via the custom element', async () => {
+    const el = /** @type {A11yInput} */ (await fixture(html`
+      <ally-input .value=${'foo'}></a11y-input>
+    `));
+
+    // add debugger
+    // debugger;
+    expect(el.value).to.equal('foo');
+  });
+});
+
+/**
+ *
+ * Adding test stub to increase `Statements` `code coverage`:
+ *
+ * =============================== Coverage summary ===============================
+ * Statements   : 78.57% ( 22/28 )
+ * Branches     : 37.5% ( 3/8 )
+ * Functions    : 87.5% ( 7/8 )
+ * Lines        : 78.57% ( 22/28 )
+ * ================================================================================
+ *
+ * 24 03 2020 11:31:30.693:ERROR [reporter.coverage-istanbul]:
+ *  Coverage for statements (78.57%) does not meet global threshold (90%)
+ * 24 03 2020 11:31:30.693:ERROR [reporter.coverage-istanbul]:
+ *  Coverage for lines (78.57%) does not meet global threshold (90%)
+ * 24 03 2020 11:31:30.693:ERROR [reporter.coverage-istanbul]:
+ *  Coverage for branches (37.5%) does not meet global threshold (90%)
+ * 24 03 2020 11:31:30.693:ERROR [reporter.coverage-istanbul]:
+ *  Coverage for functions (87.5%) does not meet global threshold (90%)
+ */
+describe('a11y input', () => {
+  it('logs "We like cats too 🐱" if the value is "cat"', async () => {
+    const el = /** @type {A11yInput} */ (await fixture(`
+      <a11y-input .value=${'cat'}></a11y-input>
+    `));
+
+      // stub console.log test
+  });
+});
+
+/**
+ * test changing `label`
+ */
+describe('a11y input', () => {
+  it('can update its label', async () => {
+    const el = /** @type {A11yInput} */ (await fixture(`
+      <a11y-input label="foo"></a11y-input>
+    `));
+
+    expect(el.label).to.equal('foo');
+
+    // change label and test
+    el.label = 'bar';
+    expect(el.label).to.equal('bar');
+  });
+});
```

Comment:
### Delta between article `code coverage` and follow-along:

![Screen Shot 2020-03-24 at 4 49 20 PM](https://user-images.githubusercontent.com/10120600/77480463-c05aaa00-6def-11ea-89b7-a19827071831.png)

Reply:
### My `code coverage` results:

<img width="947" alt="Screen Shot 2020-03-24 at 4 56 09 PM" src="https://user-images.githubusercontent.com/10120600/77480837-745c3500-6df0-11ea-8ac9-6f800934e93d.png">


### Article `code coverage` is `100%`:

![Screen Shot 2020-03-24 at 4 53 59 PM](https://user-images.githubusercontent.com/10120600/77480662-1cbdc980-6df0-11ea-80f0-9f950c187860.png)